contracts/account/AccountERC7579.sol

```diff
@@ -0,0 +1,241 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {IERC1271} from "../interfaces/IERC1271.sol";
+import {AccountBase} from "./AccountBase.sol";
+import {PackedUserOperation} from "../interfaces/IERC4337.sol";
+import {Address} from "../utils/Address.sol";
+import {IERC7579AccountConfig, IERC7579Execution, IERC7579ModuleConfig} from "../interfaces/IERC7579Account.sol";
+import {IERC7579Validator, IERC7579Module, MODULE_TYPE_VALIDATOR, MODULE_TYPE_EXECUTOR, MODULE_TYPE_FALLBACK} from "../interfaces/IERC7579Module.sol";
+import {ERC7579Utils, Mode, CallType, ExecType} from "./utils/ERC7579Utils.sol";
+import {ERC4337Utils} from "./utils/ERC4337Utils.sol";
+import {EnumerableSet} from "../utils/structs/EnumerableSet.sol";
+
+abstract contract AccountERC7579 is
+    AccountBase,
+    IERC7579ModuleConfig,
+    IERC7579Execution,
+    IERC7579AccountConfig,
+    IERC1271
+{
+    using ERC7579Utils for *;
+    using EnumerableSet for *;
+
+    EnumerableSet.AddressSet private _validators;
+    EnumerableSet.AddressSet private _executors;
+    mapping(bytes4 => address) private _fallbacks;
+
+    modifier onlyModule(uint256 moduleTypeId) {
+        _checkModule(moduleTypeId, msg.sender);
+        _;
+    }
+
+    /// @inheritdoc IERC1271
+    function isValidSignature(bytes32 hash, bytes calldata signature) public view virtual override returns (bytes4) {
+        address module = abi.decode(signature[0:20], (address));
```

Comment:
This is not standardized. Do existing modules implement this behavior? What about the off-chain "signer clients"? Do they generate this kind of prefixed signature automatically?